django/thunderstore/repository/tests/test_service_account.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+
+from thunderstore.core.factories import UserFactory
+from thunderstore.repository.models import (
+    UploaderIdentityMember,
+    UploaderIdentityMemberRole,
+)
+from thunderstore.repository.service_account import (
+    CreateServiceAccountForm,
+    DeleteServiceAccountForm,
+)
+
+
+@pytest.mark.django_db
+def test_service_account_fixture(service_account):
+    assert service_account.uuid.hex == service_account.user.username
+
+
+@pytest.mark.django_db
+def test_service_account_create(user, uploader_identity):
+    UploaderIdentityMember.objects.create(
+        user=user,
+        identity=uploader_identity,
+        role=UploaderIdentityMemberRole.owner,
+    )
+    form = CreateServiceAccountForm(user, data={"identity": uploader_identity})
+    assert form.is_valid()
+    service_account = form.save()
+    assert service_account.uuid.hex == service_account.user.username
+
+
+@pytest.mark.django_db
+def test_service_account_create_not_member(user, uploader_identity):
+    assert uploader_identity.members.filter(user=user).exists() is False
+    form = CreateServiceAccountForm(user, data={"identity": uploader_identity})
+    assert form.is_valid() is False
+    assert len(form.errors["identity"]) == 1
+    assert (
+        form.errors["identity"][0]
+        == "Select a valid choice. That choice is not one of the available choices."
+    )
+
+
+@pytest.mark.django_db
+def test_service_account_create_not_owner(user, uploader_identity):
+    UploaderIdentityMember.objects.create(
+        user=user,
+        identity=uploader_identity,
+        role=UploaderIdentityMemberRole.member,
+    )
+    form = CreateServiceAccountForm(user, data={"identity": uploader_identity})
+    assert form.is_valid() is False
+    assert len(form.errors["identity"]) == 1
+    assert (
+        form.errors["identity"][0]
+        == "Must be identity owner to create a service account"
+    )
+
+
+@pytest.mark.django_db
+def test_service_account_delete(service_account):
+    member = service_account.owner.members.first()
+    assert member.role == UploaderIdentityMemberRole.owner
+    form = DeleteServiceAccountForm(
+        member.user,
+        data={"service_account": service_account},
+    )
+    assert form.is_valid()
+    form.save()
```

Comment:
Not really sure how to finish this test (e.g. actually check the service account is deleted). `get`ting the same object using the ID and also trying to get the `User` could work, but I'm assuming there's a better way as we already have an instance, just not sure how.

Reply:
I also just realised I forgot to actually delete the `User` in `DeleteServiceAccountForm` so improving this test would help.

Reply:
There's a `.refresh_from_database` function you can use if you don't want to `get`, but I have no idea how does that behave with deleted objects.

I usually use `assert Queryset.objects.filter(pk=obj.pk).exists() is False`